### PR TITLE
Fix memory leak in maestro Cgo

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Memory allocated in C code via `malloc` or `CString` API should be explicitly freed. It will not be garbage collected by GO and not visible by pprof utility as well

### CODE REVIEW

File: processes/process_utils.c [hunk 1]
```diff
@@ -87,21 +87,29 @@ pid_t reapChildren() {
...
+// Free all the memory allocated in array elements before freeing the array itself
 void freeCStringArray(char **a) {
 	int z = 0;
 	if(a) {
 		char *s = a[z];
 		while(s) {
 			free(s);
-			z++; s = a[z];
+			z++;
+            s = a[z];
```
Comment (cmonr) [inline processes/process_utils.c]: Hmm, is this weird spacing due to '\t' vs ' ' (tabs vs spaces)?

Comment (deepikabhavnani) [reply processes/process_utils.c]: yes.. tabs vs spaces and editor configs - PR #13 is for formatting fixes

Comment (edhemphill) [reply processes/process_utils.c]: go fmt will uses tabs. Another irritation of the Golang tribe.
https://stackoverflow.com/questions/19094704/indentation-in-go-tabs-or-spaces
https://thenewstack.io/spaces-vs-tabs-a-20-year-debate-and-now-this-what-the-hell-is-wrong-with-go/

Comment (edhemphill) [reply processes/process_utils.c]: recommend you all use `go fmt` with your editor. VSCode has a good plugin for Go which will do this.

**Comment by deepikabhavnani** [comment]
Rebased on top of master to resolve conflict

**Pull request merged**